### PR TITLE
Refactor implementation for Flow 6.x

### PR DESCRIPTION
**Pull request by davidspiola**
This PR is taking care of breaking changes in Flow 6.x and should be release as a major release of the package

### CODE REVIEW

File: Classes/RoutingComponent.php [hunk 1]
```diff
@@ -34,6 +38,18 @@ class RoutingComponent extends \Neos\Flow\Mvc\Routing\RoutingComponent
...
+     * @var UriFactoryInterface
+     * @Flow\Inject
+     */
+    protected $uriFactory;
+
+    /**
+     * @var ResponseFactoryInterface
+     * @Flow\Inject
+     */
+    protected $responseFactory;
```
Comment (johannessteu) [inline Classes/RoutingComponent.php]: Do we actually need this?

File: Classes/RoutingComponent.php [hunk 2]
```diff
@@ -110,10 +128,10 @@ protected function redirectIfNecessary(ComponentContext $componentContext, UriIn
         //set default redirect statusCode if configuration is not set
         $statusCode = array_key_exists('statusCode', $this->configuration) ? $this->configuration['statusCode'] : 301;
 
-        $response = $componentContext->getHttpResponse();
-        $response->setStatus((int) $statusCode);
-        $response->setHeader('Location', (string) $uri);
+        /** @var ResponseInterface $response */
+        $response = new Response($statusCode, ['Location' => (string) $uri]);
```
Comment (johannessteu) [inline Classes/RoutingComponent.php]:
```suggestion
        $response = new Response((int) $statusCode, ['Location' => (string) $uri]);
```

**Review by johannessteu — CHANGES_REQUESTED** [review]

File: Classes/RoutingComponent.php (hunk 1 above)
Comment (davidspiola) [reply Classes/RoutingComponent.php]: well... I thought I'll use the factory. But it would be possible without as well. May I ask what your concern his?

Comment (johannessteu) [reply Classes/RoutingComponent.php]: You don't use it in the code at all, right? So it should be "useless" as it is. I'm cool with using it, but right now there is a `new Response()`

Comment (davidspiola) [reply Classes/RoutingComponent.php]: Oh.. I don't? Checking...

File: Tests/Unit/RoutingComponentTest.php [hunk 3]
```diff
@@ -86,13 +116,20 @@ public function uriWithCamelCasePathGetsModifiedToLowereCase(): void
...
         $routingComponent = new RoutingComponent();
+        $this->inject($routingComponent, 'uriFactory', $uriFactory);
 
-        $uri = new Uri('http://dev.local/äß&/');
+        $uri = $uriFactory->createUri('http://dev.local/äß&/');
         $newUri = $routingComponent->handleToLowerCase($uri);
         $newUri = $routingComponent->handleTrailingSlash($newUri);
 
-        $this->assertEquals($uri, $newUri);
+        $this->assertEquals("http://dev.local/%c3%a4%c3_&/", (string) $newUri);
```
Comment (johannessteu) [inline Tests/Unit/RoutingComponentTest.php]:
```suggestion
        $this->assertEquals($uri, $newUri);
```

**Review by johannessteu — CHANGES_REQUESTED** [review]

**Pull request merged**